src/dtx/dtx_resync.c

```diff
@@ -0,0 +1,348 @@
+/**
+ * (C) Copyright 2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * dtx: resync DTX status
+ */
+#define D_LOGFAC	DD_FAC(dtx)
+
+#include <daos/placement.h>
+#include <daos/pool_map.h>
+#include <daos_srv/vos.h>
+#include <daos_srv/pool.h>
+#include <daos_srv/dtx_srv.h>
+#include <daos_srv/daos_server.h>
+#include <daos_srv/container.h>
+#include "dtx_internal.h"
+
+struct dtx_resync_entry {
+	d_list_t		dre_link;
+	struct daos_tx_id	dre_xid;
+	daos_unit_oid_t		dre_oid;
```

Comment:
The dre_xid and dre_oid are assigned independently from the vos_iter_entry_t members. So replace them with daos_tx_entry seems not too much benefit.

Reply:
> My suggestion is to simply remove this change for this moment, that's part of reintegration work, and we don't support reintegration at all so far, why don't we leave this for reintegration design/implementation? What do you think?

Currently, we do not sync DTXs status when close the container, because we may have no chance to do that if:
1. Current server crashed, or (if we do not support re-integration, then we can ignore it temporarily)
2. Other non-leader replica may crashed or stopped already.

Since open container cannot by bypass, we can resync the DTX status instead of doing that when close. That is more easy to control. So here, the dtx_resync() when open the container is valuable.